app/views/shared/_header.html.haml

```diff
@@ -1,22 +1,22 @@
 %section#top-header.row
   %h1.col-md-2
-    = link_to "MyFLiX"
-  %ul.col-md-4.clearfix
-    %li= link_to "Videos"
-    %li= link_to "My Queue"
-    %li= link_to "People"
-  %form.col-md-5.navbar-form(action="")
-    .form-group
-      %input.form-control(type="text" placeholder="Search for videos here")
-    %button.btn.btn-default(type="submit") Search
-  #user_links.pull-right
-    %ul
-      %li.dropdown
-        %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
-          Welcome, Philip J. Fry
-          %b.caret
-        %ul.dropdown-menu(role="menu" aria-labelledby="dlabel")
-          %li
-            %a(href="#") Account
-            %a(href="#") Plan and Billing
-            %a(href="#") Sign Out
+    = link_to "MyFLiX", home_path
+  - if current_user
+    %ul.col-md-4.clearfix
+      %li= link_to "Videos", home_path
+      %li= link_to "My Queue"
+      %li= link_to "People"
+    = form_tag search_videos_path, class: "span5 for-search", method: "get" do
```

Comment:
`span5` should do nothing here as you are using Bootstrap 3 already and span grid is Bootstrap 2.